File: nodejs/messaging-api/src/routes/distributions.ts
import express, { Request, Response } from 'express';
import api from '../api/api';

const router = express.Router();

/**
 * @openapi
 * /distributions/list:
 *   get:
 *     summary: Get distribution list
 *     responses:
 *       200:
 *         description: Returns distribution list
 */
router.get('/list', (req: Request, res: Response) => {
  api.DistributionsApi.listDistributions(100, 0)
    .then((response) => {
      res.send(response.data);
    })
    .catch((error) => {
      res.status(500).send(error.response ? error.response.data : error.message);
    });
});

export default router;

File: python/messaging-api/app/resources/distributions.py
from flask_restful import Resource


class Distributions(Resource):
    def __init__(self, api_client):
        self.api_client = api_client

    def get(self):
        """
        Get distribution list
        ---
        tags:
          - Distributions
        responses:
          200:
            description: Returns distribution list
        """
        return self.api_client.get_request('/distributions'), 200
